speaker/layer2_controller.go

```diff
@@ -223,8 +223,12 @@ func poolMatchesNodeL2(pool *config.Pool, node string) bool {
 }
 
 func (c *layer2Controller) speakersForPool(l log.Logger, name string, pool *config.Pool, nodes map[string]*v1.Node) map[string]bool {
+	usableSpeakers := c.sList.UsableSpeakers()
+	if usableSpeakers == nil {
+		return nil
```

Comment:
getting back to this (sorry for the delay-2):
Here we are giving the `nil` value a semantic: it means memberlist is disabled. Can we have UsableSpeakers return a struct containing the map plus a "considerAllNodes" flag (maybe with a better name)?
At the same time, we should return that struct and use that instead of nil in the following calls.
This will give more clarity and less doubts on what the code wants to do.

Reply:
ping :-)

Reply:
I am not sure we should extend the logic. IMO the scope of the the other logic (like usableNode which I rename) seems different. Please take another look, thanks!